src/consensus/produce_block.rs

```diff
@@ -84,16 +79,13 @@ where
         Continue::Stop => true,
         Continue::Continue { .. } => false,
     };
-    (
-        Slice {
-            slot,
-            slice_index,
-            is_last,
-            merkle_root: None,
-            data,
-        },
-        cont_prod,
-    )
+    let header = SliceHeader {
+        slot,
+        slice_index,
+        is_last,
+    };
+    let payload = SlicePayload::from(data);
```

Comment:
We are decoding here only to encode later and then decode again.  This could be improved but not sure if necessarily in this PR.

Reply:
I don't have a better idea here either right now, I think we can leave it for now